src/Syntax/Resolve/Toplevel.hs

```diff
@@ -3,13 +3,12 @@ module Syntax.Resolve.Toplevel
   ( extractToplevel, extractToplevels
   ) where
 
-import Data.Triple
-
 import Syntax.Var
 import Syntax
 
 extractToplevel :: Semigroup (Var p) => Toplevel p -> ([Var p], [Var p])
-extractToplevel (LetStmt d) = (map fst3 d, [])
+extractToplevel (LetStmt d) = (map b d, []) where
+  b (Binding v _ _ _) = v
```

Comment:
I'd like to say that this is a terrible method. Would it be worth turning `Binding` into a record?

Reply:
Sure, I guess.

Reply:
Can you find the other instances of stuff like this?